v1/plugins/rest/rest_test.go

```diff
@@ -2683,3 +2736,139 @@ func newOauth2KmsClientCredentialsTestClient(t *testing.T, ts *testServer, ots *
+		if err != nil {
+			t.Fatalf("failed to unmarshal keyvault sign request = %v", err)
+		}
+
+		resp, err := json.Marshal(kvResponse{KID: "some-KID", Value: sign})
+		if err != nil {
+			t.Fatalf("json.Marshal(kvResponse{}) = %v", err)
+		}
+		w.WriteHeader(200)
+		_, err = w.Write(resp)
+		if err != nil {
+			t.Fatalf("w.Write(resp) = %v", err)
+		}
+	}))
+	defer kvServer.Close()
+
+	tokenerServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		b, err := json.Marshal(azureManagedIdentitiesToken{AccessToken: "token1"})
+		if err != nil {
+			t.Fatalf("json.Marshal(azureManagedIdentitiesToken{}) = %v", err)
+		}
+		w.WriteHeader(200)
+		_, err = w.Write(b)
+		if err != nil {
+			t.Fatalf("w.Write(b) = %v", err)
+		}
+	}))
+
+	kvURL, err := url.Parse(kvServer.URL)
+	if err != nil {
+		t.Fatalf("url.Parse(kvServer.URL) = %v", err)
+	}
+	tokenerServerURL, err := url.Parse(tokenerServer.URL)
+	if err != nil {
+		t.Fatalf("url.Parse(tokenerServer.URL) = %v", err)
+	}
+	fakeCfg := azureKeyVaultConfig{
+		URL: kvURL,
+		Alg: "ES256",
+	}
+
+	fakePlugin := &azureSigningAuthPlugin{
+		MIAuthPlugin:   &azureManagedIdentitiesAuthPlugin{Endpoint: tokenerServerURL.String()},
+		Service:        "keyvault",
+		keyVaultConfig: &fakeCfg,
+		keyVaultSignPlugin: &azureKeyVaultSignPlugin{
+			tokener: func() (string, error) { return "azure_tkn", nil },
+			config:  fakeCfg,
+		},
+	}
+
+	client := newOauth2AzureKVClient(t, &ts, &ots, tokenerServer, fakePlugin)
+	_, err = client.Do(context.Background(), http.MethodGet, "test")
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+}
+
+func newOauth2AzureKVClient(t *testing.T, ts *testServer, ots *oauth2TestServer, tkn *httptest.Server, azureSign *azureSigningAuthPlugin) *Client {
+	cfg := fmt.Sprintf(`{
+				"name": "foo",
+				"url": %q,
+				"allow_insecure_tls": true,
+				"credentials": {
+					"oauth2": {
+            "token_url": "%v/token",
+						"grant_type": %q,
+						"azure_keyvault": {
+            	"key": "tester-key",
+          		"key_algorithm": "ES256",
+          		"vault": "my-secret-kv"
+          	},
+            "azure_signing": {
+            	"service": "keyvault",
+              "azure_managed_identity": {
+								"endpoint": "%v"
+							}
+            },
+						"scopes": ["scope1", "scope2"],
+						"additional_claims": {
+							"aud": "test-audience",
+							"iss": "client-one"
+						}
+					}
+				}
+			}`, ts.server.URL, ots.server.URL, grantTypeClientCredentials, tkn.URL)
+	client, err := New([]byte(cfg), map[string]*keys.Config{})
+
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	if _, err := client.config.Credentials.OAuth2.NewClient(client.config); err != nil {
+		t.Fatalf("Oauth2.NewClient() = %q", err)
+	}
+
+	if client.config.Credentials.OAuth2.AzureSigningPlugin.keyVaultSignPlugin == nil {
+		t.Errorf("Oauth2.AzureSigningPlugin.keyVaultSignPlugin isn't setup")
+	}
+
+	// setuo fake KV config and signer
```

Comment:
```suggestion
	// setup fake KV config and signer
```

Reply:
my bad, fixed in: [4540de4](https://github.com/open-policy-agent/opa/commit/4540de41e768efc6c3ee41c1b85e8a9cf64fb74c)